src/app/frontend/scope/controller.js

```diff
@@ -0,0 +1,139 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+/**
+ * @final
+ */
+export class ScopeController {
+  /**
+   * @param {!angular.$log} $log
+   * @param {!angular.$resource} $resource
+   * @param {!angular.$q} $q
+   * @param {!angular.$timeout} $timeout
+   * @param {!angular.$window} $window
+   * @param {!backendApi.Scope} scope
+   * @param {!angular.$sce} $sce
+   * @param {!./../common/csrftoken/service.CsrfTokenService} kdCsrfTokenService
+   * @ngInject
+   */
+  constructor($log, $resource, $q, $timeout, $window, scope, $sce, kdCsrfTokenService) {
+    /** @export {!backendApi.Scope} */
+    this.scope = scope;
+
+    /** @private {!angular.$q} */
+    this.q_ = $q;
+
+    /** @private {!angular.$log} */
+    this.log_ = $log;
+
+    /** @private {!angular.$sce} */
+    this.sce_ = $sce;
+
+    /** @private {!angular.$window} */
+    this.window_ = $window;
+
+    /** @public {string} */
+    this.scope.address = String(this.sce_.trustAsResourceUrl(this.scope.address));
+
+    /** @public {boolean} */
+    this.isDeployInProgress = false;
+
+    /** @private {!angular.$timeout} */
+    this.timeout_ = $timeout;
+
+    /** @private {!angular.$resource} */
+    this.resource_ = $resource;
+
+    /** @private {!angular.$q.Promise} */
+    this.tokenPromise_ = kdCsrfTokenService.getTokenForAction('scope');
+
+    /** @public {string} */
+    this.iframeHeight = `${this.window_.innerHeight - 200} px`;
+
+    this.window_.addEventListener('resize', () => {
+      this.timeout_(() => {
+        this.iframeHeight = `${this.window_.innerHeight - 200} px`;
+      });
+    });
+  }
+
+  /**
+   * Pings until scope is up and ready.
+   * @export
+   */
+  pingScope(tries) {
+    if (tries < 5) {
```

Comment:
Calling pingScope(5) will do nothing, right? I initially was a little confused by this.
Can you refactor this argument (e.g. to decrease until 0), or add a comment about its meaning?